tests/test_events.py

```diff
@@ -275,3 +277,34 @@ def test_order_started(events_object, client, base_ddo_url):
     published_ddo = get_ddo(client, base_ddo_url, did)
     # TODO: currently the graph for v4 is WIP, need to replace this
     assert published_ddo["stats"]["consumes"] == -1
+
+
+def test_metadata_state_update(client, base_ddo_url, events_object):
+
+    web3 = events_object._web3  # get_web3()
+    block = web3.eth.block_number
+    _ddo = new_ddo(test_account1, web3, f"dt.{block}")
+    did = _ddo.id
+
+    send_create_update_tx("create", _ddo, bytes([2]), test_account1)
+    events_object.process_current_blocks()
+    published_ddo = get_ddo(client, base_ddo_url, did)
+    assert published_ddo["id"] == did
+
+    # MetadataState updated to active should delegate to MetadataCreated processor
+    send_set_metadata_state_tx(
+        ddo=_ddo, account=test_account1, state=MetadataStates.ACTIVE
+    )
+    with patch("aquarius.events.processors.MetadataCreatedProcessor.process") as mock:
+        events_object.process_current_blocks()
+        mock.assert_called()
+    published_ddo = get_ddo(client, base_ddo_url, did)
```

Comment:
the other way around, please. Delete the asset first, then test it is recreated. Do not mock the processing part, since that is the exact thing we want to test: that when state is 0, the asset is recreated. This way, you are not testing that part.